File: atlas/src/atlas-cli/src/application/shared/output/RuntimeOutputWriter.ts
/**
 * Writes intentional command results to the process output streams.
 */
export interface RuntimeOutputWriter {
  /**
   * Writes one user-facing result line to standard output.
   *
   * @param message - Complete line of command output without a trailing newline requirement.
   */
  writeLine(message: string): void;

  /**
   * Writes one user-facing error line to standard error.
   *
   * @param message - Complete line of command output without a trailing newline requirement.
   */
  writeErrorLine(message: string): void;
}

File: atlas/src/tools/cs/atlas-cs-cli/Infrastructure/RuntimeOutputWriter.cs
namespace StarCruiseStudios.Atlas.Cs.Infrastructure;

/// <summary>
/// Writes designed command results to standard output and command errors to standard error.
/// </summary>
public sealed class RuntimeOutputWriter : IRuntimeOutputWriter
{
    private readonly TextWriter output;
    private readonly TextWriter error;

    /// <summary>
    /// Creates a process output adapter from explicit streams.
    /// </summary>
    /// <param name="output">Stream used for successful command results.</param>
    /// <param name="error">Stream used for user-actionable command errors.</param>
    public RuntimeOutputWriter(TextWriter output, TextWriter error)
    {
        this.output = output;
        this.error = error;
    }

    /// <inheritdoc />
    public void WriteLine(string message) => this.output.WriteLine(message);

    /// <inheritdoc />
    public void WriteError(string message) => this.error.WriteLine(message);
}

